script/add-grammar

```diff
@@ -81,41 +89,71 @@ $url = ARGV[0]
 # No URL? Print a usage message and bail.
 unless $url
   warn usage
-  exit 1;
+  exit 1
 end
 
-# Exit early if docker isn't installed or running.
-log "Checking docker is installed and running"
-command('docker', 'ps')
+# Flags to track which changes should be reverted on an error
+$gitmodules  = File.read("#{ROOT}/.gitmodules")
+$git_config  = File.read("#{ROOT}/.git/config")
+$vendor_list = File.read("#{ROOT}/vendor/README.md")
 
-# Ensure the given URL is an HTTPS link
-parts    = parse_url $url
-https    = "https://#{parts[:host]}/#{parts[:user]}/#{parts[:repo]}"
-repo_new = "vendor/grammars/#{parts[:repo]}"
-repo_old = parse_submodule($replace) if $replace
+def restore_configs
+  File.write("#{ROOT}/.gitmodules", $gitmodules)
+  File.write("#{ROOT}/.git/config", $git_config)
+end
 
-Dir.chdir(ROOT)
+begin
+  # Exit early if Docker isn't installed or running.
+  log "Checking Docker is installed and running"
+  command('docker', 'ps')
 
-if repo_old
-  log "Deregistering: #{repo_old}"
-  command('git', 'submodule', 'deinit', repo_old)
-  command('git', 'rm', '-rf', repo_old)
-  command('script/grammar-compiler', 'update', '-f')
-end
+  # Ensure the given URL is an HTTPS link
+  parts     = parse_url $url
+  $https    = "https://#{parts[:host]}/#{parts[:user]}/#{parts[:repo]}"
+  $repo_new = "vendor/grammars/#{parts[:repo]}"
+  $repo_old = parse_submodule($replace) if $replace
```

Comment:
These shouldn't need to be global as they're defined within the `begin`

Reply:
See above. They appeared to be block-scoped or something...

I'm currently on macOS so I can't rerun the script to reproduce (owing to that same `charlock_holmes` error 😞).